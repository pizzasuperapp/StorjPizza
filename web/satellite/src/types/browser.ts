// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all properties and methods present and available in the file/browser objects in Browser.
 */
export interface BrowserFile extends File {
  Key: string;
  LastModified: Date;
  Size: number;
  type: string;
}
