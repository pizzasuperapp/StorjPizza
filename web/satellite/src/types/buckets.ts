// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all bucket-related functionality.
 */
export interface BucketsApi {
    /**
     * Fetch buckets
     *
     * @returns BucketPage
     * @throws Error
     */
    get(projectId: string, before: Date, cursor: BucketCursor): Promise<BucketPage>;

    /**
     * Fetch all bucket names
     *
     * @returns string[]
     * @throws Error
     */
    getAllBucketNames(projectId: string): Promise<string[]>;
}

/**
 * Bucket class holds info for Bucket entity.
 */
export class Bucket {
    public constructor(
        public name: string = '',
        public storage: number = 0,
        public egress: number = 0,
        public objectCount: number = 0,
        public segmentCount: number = 0,
        public since: Date = new Date(),
        public before: Date = new Date(),
    ) { }
}

/**
 * BucketPage class holds bucket total usages and flag whether more usages available.
 */
export class BucketPage {
    public constructor(
        public buckets: Bucket[] = [],
        public search: string = '',
        public limit: number = 0,
        public offset: number = 0,
        public pageCount: number = 0,
        public currentPage: number = 0,
        public totalCount: number = 0,
    ) { }
}

/**
 * BucketCursor class holds cursor for bucket name and limit.
 */
export class BucketCursor {
    public constructor(
        public search: string = '',
        public limit: number = 0,
        public page: number = 0,
    ) { }
}
