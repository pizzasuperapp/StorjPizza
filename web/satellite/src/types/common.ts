// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export enum SortDirection {
    ASCENDING = 1,
    DESCENDING,
}

export enum OnboardingOS {
    WINDOWS = "windows",
    MAC = "macos",
    LINUX = "linux",
}

export class PartneredSatellite {
    constructor(
        public name: string = '',
        public address: string = '',
    ) {}
}
