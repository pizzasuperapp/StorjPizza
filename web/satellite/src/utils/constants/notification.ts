// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export const NOTIFICATION_TYPES = {
    SUCCESS: 'SUCCESS',
    NOTIFICATION: 'NOTIFICATION',
    ERROR: 'ERROR',
    WARNING: 'WARNING',
};
