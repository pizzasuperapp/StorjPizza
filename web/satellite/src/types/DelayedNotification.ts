// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vue, {VueConstructor} from 'vue';

import { NOTIFICATION_TYPES } from '@/utils/constants/notification';
import { getId } from '@/utils/idGenerator';

import SuccessIcon from "@/../static/images/notifications/success.svg";
import NotificationIcon from "@/../static/images/notifications/notification.svg";
import ErrorIcon from "@/../static/images/notifications/error.svg";
import WarningIcon from "@/../static/images/notifications/warning.svg";

export class DelayedNotification {
    private readonly successColor: string = '#DBF1D3';
    private readonly errorColor: string = '#FFD4D2';
    private readonly infoColor: string = '#D0E3FE';
    private readonly warningColor: string = '#FCF8E3';
    public readonly id: string;

    private readonly callback: () => void;
    private timerId: ReturnType<typeof setTimeout>;
    private startTime: number;
    private remainingTime: number;

    public readonly type: string;
    public readonly message: string;
    public readonly style: { backgroundColor: string };
    public readonly icon: VueConstructor<Vue>;

    constructor(callback: () => void, type: string, message: string) {
        this.callback = callback;
        this.type = type;
        this.message = message;
        this.id = getId();
        this.remainingTime = 3000;
        this.start();

        // Switch for choosing notification style depends on notification type
        switch (this.type) {
        case NOTIFICATION_TYPES.SUCCESS:
            this.style = { backgroundColor: this.successColor };
            this.icon = SuccessIcon;
            break;

        case NOTIFICATION_TYPES.ERROR:
            this.style = { backgroundColor: this.errorColor };
            this.icon = ErrorIcon;
            break;

        case NOTIFICATION_TYPES.WARNING:
            this.style = { backgroundColor: this.warningColor };
            this.icon = WarningIcon;
            break;

        default:
            this.style = { backgroundColor: this.infoColor };
            this.icon = NotificationIcon;
            break;
        }
    }

    public pause(): void {
        clearTimeout(this.timerId);
        this.remainingTime -= new Date().getMilliseconds() - this.startTime;
    }

    public start(): void {
        this.startTime = new Date().getMilliseconds();
        this.timerId = setTimeout(this.callback, this.remainingTime);
    }
}
