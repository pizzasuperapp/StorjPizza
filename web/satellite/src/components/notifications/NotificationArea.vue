// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div v-if="doNotificationsExist" class="notification-container">
        <NotificationItem
            v-for="notification in notifications"
            :key="notification.id"
            :notification="notification"
        />
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import NotificationItem from '@/components/notifications/NotificationItem.vue';

import { DelayedNotification } from '@/types/DelayedNotification';

// @vue/component
@Component({
    components: {
        NotificationItem,
    },
})
export default class NotificationArea extends Vue {
    /**
     * Returns all notification queue from store.
     */
    public get notifications(): DelayedNotification[] {
        return this.$store.state.notificationsModule.notificationQueue;
    }

    /**
     * Indicates if any notifications are in queue.
     */
    public get doNotificationsExist(): boolean {
        return this.notifications.length > 0;
    }
}
</script>

<style scoped lang="scss">
    .notification-container {
        width: 417px;
        background-color: transparent;
        display: flex;
        flex-direction: column;
        position: fixed;
        top: 114px;
        right: 17px;
        align-items: flex-end;
        justify-content: space-between;
        border-radius: 12px;
        z-index: 9999;
        overflow: hidden;
    }
</style>
