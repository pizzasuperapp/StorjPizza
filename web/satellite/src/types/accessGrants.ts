// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import { SortDirection } from '@/types/common';

export type OnHeaderClickCallback = (sortBy: AccessGrantsOrderBy, sortDirection: SortDirection) => Promise<void>;

/**
 * AccessGrantsWorker provides access to the WASM module.
 */
export interface AccessGrantsWorkerFactory {
    // TODO: this should be converted to a proper interface.
    create(): Worker;
}

/**
 * Exposes all access grants-related functionality.
 */
export interface AccessGrantsApi {
    /**
     * Fetch accessGrants
     *
     * @returns AccessGrantsPage
     * @throws Error
     */
    get(projectId: string, cursor: AccessGrantCursor): Promise<AccessGrantsPage>;

    /**
     * Create new accessGrant
     *
     * @returns AccessGrant
     * @throws Error
     */
    create(projectId: string, name: string): Promise<AccessGrant>;

    /**
     * Delete existing access grant
     *
     * @returns null
     * @throws Error
     */
    delete(ids: string[]): Promise<void>;

    /**
     * Delete existing access grant by name and project id
     *
     * @returns null
     * @throws Error
     */
    deleteByNameAndProjectID(name: string, projectID: string): Promise<void>;

    /**
     * Get gateway credentials using access grant
     *
     * @returns EdgeCredentials
     * @throws Error
     */
    getGatewayCredentials(accessGrant: string, optionalURL?: string, isPublic?: boolean): Promise<EdgeCredentials>;
}

/**
 * Holds access grants sorting parameters.
 */
export enum AccessGrantsOrderBy {
    NAME = 1,
    CREATED_AT,
}

/**
 * AccessGrantCursor is a type, used to describe paged access grants list.
 */
export class AccessGrantCursor {
    public constructor(
        public search: string = '',
        public limit: number = 6,
        public page: number = 1,
        public order: AccessGrantsOrderBy = AccessGrantsOrderBy.NAME,
        public orderDirection: SortDirection = SortDirection.ASCENDING,
    ) {}
}

/**
 * AccessGrantsPage is a type, used to describe paged access grants list.
 */
export class AccessGrantsPage {
    public constructor(
        public accessGrants: AccessGrant[] = [],
        public search: string = '',
        public order: AccessGrantsOrderBy = AccessGrantsOrderBy.NAME,
        public orderDirection: SortDirection = SortDirection.ASCENDING,
        public limit: number = 6,
        public pageCount: number = 0,
        public currentPage: number = 1,
        public totalCount: number = 0,
    ) {}
}

/**
 * AccessGrant class holds info for Access Grant entity.
 */
export class AccessGrant {
    public isSelected: boolean;

    constructor(
        public id: string = '',
        public name: string = '',
        public createdAt: Date = new Date(),
        public secret: string = '',
    ) {
        this.isSelected = false;
    }

    /**
     * Returns created date as a local string.
     */
    public localDate(): string {
        return this.createdAt.toLocaleString('en-US', {timeZone: 'UTC', timeZoneName: 'short'});
    }
}

/**
 * DurationPermission class holds info for access grant's duration permission.
 */
export class DurationPermission {
    constructor(
        public notBefore: Date | null = null,
        public notAfter: Date | null = null,
    ) {}
}

/**
 * EdgeCredentials class holds info for edge credentials generated from access grant.
 */
export class EdgeCredentials {
    constructor(
        public id: string = '',
        public createdAt: Date = new Date(),
        public accessKeyId: string = '',
        public secretKey: string = '',
        public endpoint: string = '',
    ) {}
}
