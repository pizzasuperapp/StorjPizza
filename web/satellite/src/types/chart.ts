// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

import { DataStamp } from "@/types/projects";
import { Size } from "@/utils/bytesSize";

/**
 * ChartData class holds info for ChartData entity.
 */
export class ChartData {
    public labels: string[];
    public datasets: DataSets[] = [];

    public constructor(
        labels: string[],
        backgroundColor: string,
        borderColor: string,
        pointBorderColor: string,
        data: number[],
        secondaryBackgroundColor?: string,
        secondaryBorderColor?: string,
        secondaryPointBorderColor?: string,
        secondaryData?: number[],
    ) {
        this.labels = labels;
        this.datasets[0] = new DataSets(backgroundColor, borderColor, pointBorderColor, data)

        if (secondaryData && secondaryBackgroundColor && secondaryBorderColor && secondaryPointBorderColor) {
            this.datasets[1] = new DataSets(secondaryBackgroundColor, secondaryBorderColor, secondaryPointBorderColor, secondaryData);
        }
    }
}

/**
 * DataSets class holds info for chart's DataSets entity.
 */
class DataSets {
    public constructor(
        public backgroundColor: string,
        public borderColor: string,
        public pointBorderColor: string,
        public data: number[],
        public borderWidth: number = 4,
        public pointHoverBackgroundColor: string = 'white',
        public pointHoverBorderWidth: number = 5,
    ) {}
}

/**
 * TooltipParams holds tooltip's configuration
 */
export class TooltipParams {
    public constructor(
        public tooltipModel: TooltipModel,
        public chartId: string,
        public tooltipId: string,
        public markUp: string,
        public tooltipTop: number,
        public tooltipLeft: number,
    ) {}
}

/**
 * StylingConstants holds tooltip styling constants
 */
class StylingConstants {
    public static tooltipOpacity = '1';
    public static tooltipPosition = 'absolute';
}

/**
 * Styling holds tooltip's styling configuration
 */
class Styling {
    public constructor(
        public tooltipModel: TooltipModel,
        public element: HTMLElement,
        public topPosition: number,
        public leftPosition: number,
        public chartPosition: ClientRect,
    ) {}
}

/**
 * Color is a color definition.
 */
export type Color = string

/**
 * TooltipItem contains datapoint information.
 */
export interface TooltipItem {
    // Label for the tooltip
    label: string,

    // Value for the tooltip
    value: string,

    // X Value of the tooltip
    // (deprecated) use `value` or `label` instead
    xLabel: number | string,

    // Y value of the tooltip
    // (deprecated) use `value` or `label` instead
    yLabel: number | string,

    // Index of the dataset the item comes from
    datasetIndex: number,

    // Index of this data item in the dataset
    index: number,

    // X position of matching point
    x: number,

    // Y position of matching point
    y: number
}

/**
 * TooltipModel contains parameters that can be used to render the tooltip.
 */
export interface TooltipModel {
    // The items that we are rendering in the tooltip. See Tooltip Item Interface section
    dataPoints: TooltipItem[],

    // Positioning
    xPadding: number,
    yPadding: number,
    xAlign: string,
    yAlign: string,

    // X and Y properties are the top left of the tooltip
    x: number,
    y: number,
    width: number,
    height: number,
    // Where the tooltip points to
    caretX: number,
    caretY: number,

    // Body
    // The body lines that need to be rendered
    // Each object contains 3 parameters
    // before: string[] // lines of text before the line with the color square
    // lines: string[], // lines of text to render as the main item with color square
    // after: string[], // lines of text to render after the main lines
    body: {before: string[]; lines: string[], after: string[]}[],
    // lines of text that appear after the title but before the body
    beforeBody: string[],
    // line of text that appear after the body and before the footer
    afterBody: string[],
    bodyFontColor: Color,
    _bodyFontFamily: string,
    _bodyFontStyle: string,
    _bodyAlign: string,
    bodyFontSize: number,
    bodySpacing: number,

    // Title
    // lines of text that form the title
    title: string[],
    titleFontColor: Color,
    _titleFontFamily: string,
    _titleFontStyle: string,
    titleFontSize: number,
    _titleAlign: string,
    titleSpacing: number,
    titleMarginBottom: number,

    // Footer
    // lines of text that form the footer
    footer: string[],
    footerFontColor: Color,
    _footerFontFamily: string,
    _footerFontStyle: string,
    footerFontSize: number,
    _footerAlign: string,
    footerSpacing: number,
    footerMarginTop: number,

    // Appearance
    caretSize: number,
    caretPadding: number,
    cornerRadius: number,
    backgroundColor: Color,

    // colors to render for each item in body[]. This is the color of the squares in the tooltip
    labelColors: Color[],
    labelTextColors: Color[],

    // 0 opacity is a hidden tooltip
    opacity: number,
    legendColorBackground: Color,
    displayColors: boolean,
    borderColor: Color,
    borderWidth: number
}

/**
 * Tooltip provides custom tooltip rendering
 */
export class Tooltip {
    public static custom(params: TooltipParams): void {
        const chart = document.getElementById(params.chartId);

        if (!chart) {
            return;
        }

        const tooltip: HTMLElement = Tooltip.createTooltip(params.tooltipId);

        if (!params.tooltipModel.opacity) {
            Tooltip.remove(tooltip);

            return;
        }

        if (params.tooltipModel.body) {
            Tooltip.render(tooltip, params.markUp);
        }

        const position = chart.getBoundingClientRect();

        const tooltipStyling = new Styling(params.tooltipModel, tooltip, params.tooltipTop, params.tooltipLeft, position);
        Tooltip.elemStyling(tooltipStyling);
    }

    public static createTooltip(id: string): HTMLElement {
        let tooltipEl = document.getElementById(id);

        if (!tooltipEl) {
            tooltipEl = document.createElement('div');
            tooltipEl.id = id;
            document.body.appendChild(tooltipEl);
        }

        return tooltipEl;
    }

    public static remove(tooltipEl: HTMLElement): void {
        document.body.removeChild(tooltipEl);
    }

    private static render(tooltip: HTMLElement, markUp: string): void {
        tooltip.innerHTML = markUp;
    }

    private static elemStyling(elemStyling: Styling): void {
        elemStyling.element.style.opacity = StylingConstants.tooltipOpacity;
        elemStyling.element.style.position = StylingConstants.tooltipPosition;
        elemStyling.element.style.left = `${elemStyling.chartPosition.left + elemStyling.tooltipModel.caretX - elemStyling.leftPosition}px`;
        elemStyling.element.style.top = `${elemStyling.chartPosition.top + window.pageYOffset + elemStyling.tooltipModel.caretY - elemStyling.topPosition}px`;
    }
}

/**
 * Stores data for chart's tooltip
 */
export class ChartTooltipData {
    public date: string;
    public value: string;

    public constructor(stamp: DataStamp) {
        const size = new Size(stamp.value, 1)

        this.date = stamp.intervalStart.toLocaleDateString('en-US', { day: '2-digit', month: 'short' });
        this.value = `${size.formattedBytes} ${size.label}`;
    }
}

/**
 * RenderChart contains definition for renderChart and addPlugin, that can be used to cast
 * a derived chart type, with `(this as unknown as RenderChart).renderChart`
 */
export interface RenderChart {
    renderChart<A, B>(A, B): void
    addPlugin (plugin?: Record<string, (chart) => void>): void
}
