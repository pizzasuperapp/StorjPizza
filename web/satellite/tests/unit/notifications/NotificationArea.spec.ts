// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import NotificationArea from '@/components/notifications/NotificationArea.vue';

import { DelayedNotification } from '@/types/DelayedNotification';
import { NOTIFICATION_TYPES } from '@/utils/constants/notification';
import { mount, shallowMount } from '@vue/test-utils';

describe('NotificationArea.vue', () => {
    it('renders correctly', () => {
        const wrapper = shallowMount(NotificationArea, {
            computed: {
                notifications: () => [],
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with notification', () => {
        const testMessage = 'testMessage';
        const notifications = [new DelayedNotification(
            jest.fn(),
            NOTIFICATION_TYPES.SUCCESS,
            testMessage,
        ), new DelayedNotification(
            jest.fn(),
            NOTIFICATION_TYPES.ERROR,
            testMessage,
        ), new DelayedNotification(
            jest.fn(),
            NOTIFICATION_TYPES.WARNING,
            testMessage,
        ), new DelayedNotification(
            jest.fn(),
            NOTIFICATION_TYPES.NOTIFICATION,
            testMessage,
        )];

        const wrapper = mount(NotificationArea, {
            computed: {
                notifications: () => notifications,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
