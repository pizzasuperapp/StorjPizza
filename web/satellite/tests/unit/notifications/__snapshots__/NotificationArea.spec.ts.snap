// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationArea.vue renders correctly 1`] = ``;

exports[`NotificationArea.vue renders correctly with notification 1`] = `
<div class="notification-container">
  <div class="notification-wrap" style="background-color: rgb(219, 241, 211);">
    <div class="notification-wrap__content-area">
      <div class="notification-wrap__content-area__image"><svg></svg></div>
      <div class="notification-wrap__content-area__message-area">
        <p class="notification-wrap__content-area__message">testMessage</p>
        <!---->
      </div>
    </div>
    <div class="notification-wrap__buttons-group"><span class="notification-wrap__buttons-group__close"><svg></svg></span></div>
  </div>
  <div class="notification-wrap" style="background-color: rgb(255, 212, 210);">
    <div class="notification-wrap__content-area">
      <div class="notification-wrap__content-area__image"><svg></svg></div>
      <div class="notification-wrap__content-area__message-area">
        <p class="notification-wrap__content-area__message">testMessage</p>
        <!---->
      </div>
    </div>
    <div class="notification-wrap__buttons-group"><span class="notification-wrap__buttons-group__close"><svg></svg></span></div>
  </div>
  <div class="notification-wrap" style="background-color: rgb(252, 248, 227);">
    <div class="notification-wrap__content-area">
      <div class="notification-wrap__content-area__image"><svg></svg></div>
      <div class="notification-wrap__content-area__message-area">
        <p class="notification-wrap__content-area__message">testMessage</p>
        <!---->
      </div>
    </div>
    <div class="notification-wrap__buttons-group"><span class="notification-wrap__buttons-group__close"><svg></svg></span></div>
  </div>
  <div class="notification-wrap" style="background-color: rgb(208, 227, 254);">
    <div class="notification-wrap__content-area">
      <div class="notification-wrap__content-area__image"><svg></svg></div>
      <div class="notification-wrap__content-area__message-area">
        <p class="notification-wrap__content-area__message">testMessage</p>
        <!---->
      </div>
    </div>
    <div class="notification-wrap__buttons-group"><span class="notification-wrap__buttons-group__close"><svg></svg></span></div>
  </div>
</div>
`;
